grading/base/grading/projects.py

```
@@ -143,36 +143,41 @@ def __init__(self, main_file_name='Main.java', source='1.7', sourcepath="src",
             of the file where the code will be stored.
         """
 
-        self._main_file_name = main_file_name
+        self._main_class = main_class
+        self._main_file_name = self._main_class + ".java"
         self._source = source
         self._sourcepath = sourcepath
         self._classpath = classpath
 
     def create_from_code(self, code):
-        def run(input_file):
-            javac_command = ["javac", "-source", self._source, self._main_file_name]
-            _run_in_sandbox(javac_command, stdin=input_file, cwd=CODE_WORKING_DIR)
-
-            java_command = ["java", self._main_file_name.replace(".java", "")]
-            return _run_in_sandbox(java_command, stdin=input_file, cwd=CODE_WORKING_DIR)
+        project_directory = tempfile.mkdtemp(dir=CODE_WORKING_DIR)
+        source_directory = os.path.join(project_directory, self._sourcepath)
+        if not os.path.exists(source_directory):
```

This is confusing. This directory should not exist before (you're using a a temporary directory). I think this case is unnecessary and misleading (if it does exist, which should not, it's better to fail fast: let it throw an exception and we'll know we have a problem)